src/EnergyPlus/SimAirServingZones.cc

```diff
@@ -2070,98 +1997,127 @@ void InitAirLoops(EnergyPlusData &state, bool const FirstHVACIteration) // TRUE
+                thisPrimaryAirSys.supFanVecIndex = SupFanIndex;
+                thisPrimaryAirSys.supFanModelType = ObjectVectorOOFanSystemModel;
             }
             if (FoundCentralCoolCoil) { // parent systems with fan will need to set the fan placement
-                PrimaryAirSystems(AirLoopNum).supFanLocation = FanPlacement::DrawThru;
+                thisPrimaryAirSys.supFanLocation = FanPlacement::DrawThru;
             } else {
-                PrimaryAirSystems(AirLoopNum).supFanLocation = FanPlacement::BlowThru;
+                thisPrimaryAirSys.supFanLocation = FanPlacement::BlowThru;
             }
 
             if (retFanModelType == StructArrayLegacyFanModels) {
-                PrimaryAirSystems(AirLoopNum).retFanModelType = StructArrayLegacyFanModels;
-                PrimaryAirSystems(AirLoopNum).RetFanNum = RetFanIndex;
+                thisPrimaryAirSys.retFanModelType = StructArrayLegacyFanModels;
+                thisPrimaryAirSys.RetFanNum = RetFanIndex;
             } else if (retFanModelType == ObjectVectorOOFanSystemModel) {
-                PrimaryAirSystems(AirLoopNum).retFanModelType = ObjectVectorOOFanSystemModel;
-                PrimaryAirSystems(AirLoopNum).retFanVecIndex = RetFanIndex;
+                thisPrimaryAirSys.retFanModelType = ObjectVectorOOFanSystemModel;
+                thisPrimaryAirSys.retFanVecIndex = RetFanIndex;
             }
         }
         // Check whether there are Central Heating Coils in the Primary Air System
-        for (AirLoopNum = 1; AirLoopNum <= NumPrimaryAirSys; ++AirLoopNum) {
+        for (int AirLoopNum = 1; AirLoopNum <= numPrimaryAirSys; ++AirLoopNum) {
+            auto &thisPrimaryAirSys = state.dataAirSystemsData->PrimaryAirSystems(AirLoopNum);
             bool FoundCentralHeatCoil = false;
-            for (BranchNum = 1; !FoundCentralHeatCoil && BranchNum <= PrimaryAirSystems(AirLoopNum).NumBranches; ++BranchNum) {
-                for (CompNum = 1; !FoundCentralHeatCoil && CompNum <= PrimaryAirSystems(AirLoopNum).Branch(BranchNum).TotalComponents; ++CompNum) {
-                    CompTypeNum = PrimaryAirSystems(AirLoopNum).Branch(BranchNum).Comp(CompNum).CompType_Num;
-                    if (CompTypeNum == CompType::WaterCoil_SimpleHeat || CompTypeNum == CompType::Coil_ElectricHeat ||
-                        CompTypeNum == CompType::Coil_GasHeat || CompTypeNum == CompType::SteamCoil_AirHeat ||
-                        CompTypeNum == CompType::Coil_DeSuperHeat || CompTypeNum == CompType::DXHeatPumpSystem ||
-                        CompTypeNum == CompType::Furnace_UnitarySys_HeatOnly || CompTypeNum == CompType::Furnace_UnitarySys_HeatCool ||
-                        CompTypeNum == CompType::UnitarySystem_BypassVAVSys || CompTypeNum == CompType::UnitarySystem_MSHeatPump ||
-                        CompTypeNum == CompType::CoilUserDefined) {
+            bool unitaryCoolingCoilExists = false;
+            bool unitaryHeatingCoilExists = false;
```

Comment:
I wondered whether these should be declared outside the loop scope where they are used. Since there are 2 loops (lines 2026 - 2093) that use these maybe it is better to declare once and be done.